mantidproject/mantid
# Pychop fixes

**Description of work.**

Fixes various bugs in PyChop identified in #23738 and also updates the measured flux table for MARI after that instrument's guide upgrade.

**Report to:** [nobody]. 

**To test:**



1. Check that when plotting, the graph in the Q-E tab still has its axis labels.
2. Select any `LET` as the instrument, put `1` in the `Ei` edit box and check the `Show multi-reps` box. Click `Show data ascii window` and make sure that there are more than one Ei shown there (the focus Ei (1meV) should be first followed by the other reps ).
3. Click `Calculate and Plot` and see that the plot yields a range of different Ei. Replace the `1` in the `Ei` box with the energy of one of the other reps shown in the plot (e.g. `1.40` or `19.82`). Click `Calculate and Plot` again and check that all the different energies of reps are the same (they will not be in the same order though). Alternatively, switch to the `Time-Distance` tab and check that the plot does not change when you change the `Ei` to another value in the list of reps.

Fixes #23738 



Release notes to follow.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
